index.js

```diff
@@ -66,152 +62,109 @@ function releaseConnectionToPool(release) {
 
 function decorateWithQueryRowsAsync(env, client) {
   return Object.assign(client, {
-    queryRowsAsync: (query, args) => queryWithCtxAsync(env, client, query, args).then(getRows)
+    queryRowsAsync: (query, args) => client.queryAsync(query, args).then(res => res.rows)
   })
 }
 
-function executeQueryRowsAsync(env, connector, query, args) {
-  return using(getConnection(env, connector), function (connection) {
-    return connection.queryRowsAsync(query, args)
-  })
+function executeQueryRows(env, connector, query, args) {
+  return using(getConnection(env, connector), connection =>
+    connection.queryRowsAsync(query, args)
+  )
 }
 
-function getRows(res) {
-  return res.rows
-}
-
-function queryWithCtxAsync(env, client, query, args) {
-  if (_.isObject(query) && query[env.queryValuesKey] && Array.isArray(args) && args.length > 0) {
-    throw new Error('Both query.values and args were passed to query. Please use only one of them.')
-  }
-
-  return client.queryAsync(query[env.queryTextKey] || query, query[env.queryValuesKey] || args)
+function executeQuery(env, connector, query, args) {
+  return using(getConnection(env, connector), connection =>
+    connection.queryAsync(query, args)
+  )
 }
 
 function constructLockingBeginStatement(involvedTables) {
-  var lockSql = 'LOCK TABLE {table} IN SHARE ROW EXCLUSIVE MODE'
-  var statements = involvedTables.map(function(table) {
-    return stringTemplate(lockSql, { table: table })
-  })
-  statements.unshift('BEGIN')
-  return statements.join(';')
+  const statements = involvedTables.map(table =>
```

Comment:
Couldn't this be achieved with the `LOCK TABLE a, b, ... z IN SHARE ROW EXCLUSIVE MODE` syntax?

According to the [docs](https://www.postgresql.org/docs/current/sql-lock.html),

> The command `LOCK TABLE a, b;` is equivalent to` LOCK TABLE a; LOCK TABLE b;`. The tables are locked one-by-one in the order specified in the LOCK TABLE command.

Reply:
(I don't think is that important, but I suppose it just could be optimized a bit to avoid unnecessary round-trip times if multiple tables are being locked )

Reply:
Good point. It's not obvious to me that there would be multiple round trips and I realised there's no test for multiple table locking, which would need to be added. The generating code could, however, be a bit cleaner if the change was done. I suggest to put this into a future issue, where the test case would also be included.

Reply:
I forgot what the code was doing. There won't be multiple round trips.